metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -78,50 +78,62 @@ final class DefinitionProvider(
   val scaladocDefinitionProvider =
     new ScaladocDefinitionProvider(buffers, trees, destinationProvider)
 
+  private def isAmmonnite(path: AbsolutePath): Boolean =
+    path.isAmmoniteScript && buildTargets
+      .inverseSources(path)
+      .flatMap(buildTargets.targetData)
+      .exists(_.isAmmonite)
+
   def definition(
       path: AbsolutePath,
       params: TextDocumentPositionParams,
       token: CancelToken,
-  ): Future[DefinitionResult] = {
-    val fromSemanticdb =
-      semanticdbs().textDocument(path).documentIncludingStale
-    val fromSnapshot = fromSemanticdb match {
-      case Some(doc) =>
-        definitionFromSnapshot(path, params, doc)
-      case _ =>
-        DefinitionResult.empty
-    }
-    val fromCompilerOrSemanticdb =
-      fromSnapshot match {
-        case defn if defn.isEmpty && path.isScalaFilename =>
+  ): Future[DefinitionResult] =
+    for {
+      fromCompiler <-
+        if (path.isScalaFilename && !isAmmonnite(path))
```

Comment:
Could we just keep the current order for Ammonite? With this change we would not use compiler for Ammonite, right?

I wonder if we could just adjust tests to take that into account?

Reply:
I can also fix it and map generated Ammonite files to source files like we do for Scala CLI, if we want to still have Ammonite support.

Reordering is only half of a fix since for local symbols using pc will lead to generated sources. But I can also do that, if don't know where we are with Ammonite support.

Reply:
I raised the topic on discord again and will start pushing it. In the meantime I don't think we want to work on any fixes, which we will be removing in the future.

So if it's too much work let's just leave it as is then.